src/main/java/com/coveo/spillway/limit/Limit.java

```diff
@@ -78,12 +95,33 @@ public Duration getExpiration() {
     return definition.getExpiration();
   }
 
+  public Duration getExpiration(T context) {
+    return findLimitOverride(context).map(p -> p.getExpiration()).orElse(getExpiration());
+  }
+
   public int getCapacity() {
     return definition.getCapacity();
   }
 
+  public int getCapacity(T context) {
+    return findLimitOverride(context).map(p -> p.getCapacity()).orElse(getCapacity());
+  }
+
+  public Set<LimitOverride> getLimitOverrides() {
+    return limitOverrides;
+  }
+
   @Override
   public String toString() {
     return definition.toString();
   }
-}
+
+  private Optional<LimitOverride> findLimitOverride(T context) {
+    String property = getProperty(context);
+
+    return limitOverrides
+        .stream()
+        .filter(override -> override.getProperty().equals(property))
```

Comment:
I don't think we can expect many overrides, but yes that would be a good thing to do.
Still I would not do it in this PR, I would open a new story for optimization.